packages/plugin-session-replay-browser/src/helpers.ts

```diff
@@ -0,0 +1,25 @@
+import { Event, IdentifyOperation } from '@amplitude/analytics-types';
+import { PROPERTY_ADD_OPERATIONS } from './constants';
+
+export const parseUserProperties = (event: Event) => {
+  if (!event.user_properties) {
+    return;
+  }
+  let userPropertiesObj = {};
+  const userPropertyKeys = Object.keys(event.user_properties);
+
+  userPropertyKeys.forEach((identifyKey) => {
+    if (
+      Object.values(IdentifyOperation).includes(identifyKey as IdentifyOperation) &&
+      PROPERTY_ADD_OPERATIONS.includes(identifyKey as IdentifyOperation)
+    ) {
+      const typedUserPropertiesOperation =
+        event.user_properties && (event.user_properties[identifyKey as IdentifyOperation] as Record<string, any>);
```

Comment:
We're checking to see if its defined outside of the function, so TS is saying "yeah you checked if it was defined at the time this function was created, but when this function is running, the value of event.user_properties could've changed since the function was created". I'd need to add an ```if (!event.user_properties) {
    return;
  }```
at the top of the forEach function to not have to do the check here as well.

There is also a typing issue with the `user_properties` object - I know at this point that it will definitely be a key that is one of the IdentifyOperations, but the type is written so that it could just be a generic string key as well, with a value of any. I believe this is actually not possible based on the documentation for the $identify event, but I'm not going to change the type definition right now, and our checks ensure that we are looking for one of the IdentifyOperations anyways